src/containers/audio-rewards-page/components/ClaimRewardButton.tsx

```diff
@@ -1,22 +1,144 @@
-import React from 'react'
+import React, { useEffect, useCallback } from 'react'
 
-import { useSelector } from 'react-redux'
+import { Button, ButtonType } from '@audius/stems'
+import { useDispatch, useSelector } from 'react-redux'
 
+import { useModalState } from 'hooks/useModalState'
 import { useScript } from 'hooks/useScript'
 import AudiusBackend from 'services/AudiusBackend'
-import { getUserHandle } from 'store/account/selectors'
+import { getAccountUser, getUserHandle } from 'store/account/selectors'
 import { COGNITO_SCRIPT_URL } from 'utils/constants'
+import { encodeHashId } from 'utils/route/hashIds'
 
+import {
+  ClaimStatus,
+  CognitoFlowStatus,
+  getHCaptchaStatus,
+  HCaptchaStatus,
+  setClaimStatus,
+  setCognitoFlowStatus
+} from '../store/slice'
+import {
+  ChallengeRewardID,
+  FailureReason,
+  FlowErrorEvent,
+  FlowSessionEvent,
+  FlowUICloseEvent,
+  FlowUIOpenEvent
+} from '../types'
+
+import styles from './modals/ChallengeRewards.module.css'
+
+const messages = {
+  claimYourReward: 'Claim Your Reward'
+}
+
+type ClaimRewardButtonProps = {
+  challengeId: ChallengeRewardID
+  specifier: string
+  amount: number
+  isDisabled: boolean
+  icon: JSX.Element
+  className?: string
+}
+
+// TODO: Closing modal (meaning this button no longer on screen) should reset both claim and hcaptcha statuses
+// =============
 // this button will handle triggering the cognito flow
 // or showing the HCaptchaModal
 // or both
 // logic for which flow exactly to trigger is TBD
 // pending AAO changes and DP undisbursed challenges integration
-const ClaimRewardButton = () => {
+const ClaimRewardButton = ({
+  challengeId,
+  specifier,
+  amount,
+  isDisabled,
+  icon,
+  className
+}: ClaimRewardButtonProps) => {
   const handle = useSelector(getUserHandle)
+  const currentUser = useSelector(getAccountUser)
+  const hCaptchaStatus = useSelector(getHCaptchaStatus)
+  const dispatch = useDispatch()
   const scriptLoaded = useScript(COGNITO_SCRIPT_URL)
+  const [, setHCaptchaOpen] = useModalState('HCaptcha')
+  const [, setRewardModalOpen] = useModalState('ChallengeRewardsExplainer')
 
-  const handleClick = async () => {
+  const claimReward = useCallback(async () => {
+    dispatch(setClaimStatus({ status: ClaimStatus.CLAIMING }))
+
+    const currentUserId = currentUser?.user_id ?? 0
+    const recipientEthAddress = currentUser?.wallet ?? null
+    if (!currentUserId || !recipientEthAddress) {
+      throw new Error('user id or wallet not available')
+    }
+
+    const encodedUserId = encodeHashId(currentUserId)
+    // TODO: should these be in env var or remote config?
+    const quorumSize = 2
+    const oracleEthAddress = '0xEc3a6aad822630a37210531411A4CD625EC59b33'
+    const AAOEndpoint = 'http://34.83.140.15:8000'
+
+    const response = await AudiusBackend.submitAndEvaluateAttestations({
+      challengeId,
+      encodedUserId,
+      handle,
+      recipientEthAddress, // TODO: double-check this, is it correct?
```

Comment:
the wallet is the same as recipient's eth address right? any differences in letter casing?